src/main/scala/gov/uk/justice/digital/offenderpollpush/injection/AwsSnsPublisher.java

```diff
@@ -0,0 +1,81 @@
+
+import akka.Done;
+import akka.actor.ActorSystem;
+import akka.stream.ActorMaterializer;
+import akka.stream.Materializer;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import com.github.matsluni.akkahttpspi.AkkaHttpClient;
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import java.net.URI;
+import java.util.concurrent.CompletionStage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sns.SnsAsyncClient;
+import software.amazon.awssdk.services.sns.model.PublishRequest;
+
+public class AwsSnsPublisher {
+    private final static Logger log = LoggerFactory.getLogger(AwsSnsPublisher.class);
+
+    private final String region;
+    private final URI uri;
+    private final String topicArn;
+    private final String awsAccessKeyId;
+    private final String awsSecretAccessKey;
+
+    private final Materializer materializer;
+    private final SdkAsyncHttpClient httpClient;
+    private final ActorSystem system;
+
+    public @Inject
+    AwsSnsPublisher(@Named("snsEndpoint") final String snsEndpoint,
+                    @Named("snsArnTopic") final String snsArnTopic,
+                    @Named("snsRegion") final String snsRegion,
+                    @Named("awsAccessKeyId") final String awsAccessKeyId,
+                    @Named("awsSecretAccessKey") final String awsSecretAccessKey) {
+
+        this.system = ActorSystem.create("AwsSnsPublisher");
+        this.materializer = ActorMaterializer.create(system);
+        this.uri = URI.create(snsEndpoint);
+        this.region = snsRegion;
+        this.topicArn = snsArnTopic;
+        this.awsAccessKeyId = awsAccessKeyId;
+        this.awsSecretAccessKey = awsSecretAccessKey;
+
+        this.httpClient = AkkaHttpClient.builder().withActorSystem(system).build();
+    }
+
+    public void run(final String offenderJson) {
+
+        log.info("PUBLISH TO SNS {}", offenderJson);
+
+        final SnsAsyncClient snsAsyncClient =
+            SnsAsyncClient.builder()
+                .credentialsProvider(
+                    StaticCredentialsProvider.create(AwsBasicCredentials.create(awsAccessKeyId, awsSecretAccessKey)))
+                .httpClient(httpClient)
+                .endpointOverride(uri)
+                .region(Region.of(region))
+                .build();
+
+        publishToSourceTopicWithFlow(snsAsyncClient, offenderJson).thenAccept(done -> snsAsyncClient.close());
+
+        system.registerOnTermination(snsAsyncClient::close);
+    }
+
+    private CompletionStage<Done> publishToSourceTopicWithFlow(final SnsAsyncClient sqsClient, final String msgJson) {
+        return Source.single(PublishRequest.builder()
```

Comment:
Maybe we need to model the event body more like NOMIS, example here:

`"{\"eventType\":\"EXTERNAL_MOVEMENT_RECORD-INSERTED\",\"eventDatetime\":\"2020-01-13T11:33:23.790725\",\"bookingId\":1200835,\"movementSeq\":1,\"nomisEventType\":\"M1_RESULT\"}"`

If we did we would have an eventType of say `OFFENDER_CHANGED` and maybe `offenderId=123` and `crn=A12345T` rather than the actual body of the offender. This forces clients to always get the latest state and not rely on the body.

Be interested to know what @markberridge and @Mjwillis thinks?

Reply:
Agree, message should be very minimal, no data in the payload other than ID

Reply:
Agreed, I think we should align with the NOMIS format to make it simpler to share consumers should we want to.  Do we need to add an attribute to indicate the source of the event?

Reply:
@markberridge not a bad idea, if you are using the same topic, but I think we said were not?

Reply:
As per chats on slack, I have added source, event type and subject (all configurable). The message also now has CRN and offender ID, unpacked from existing JSON.